packages/web/src/flash/FlashReducer.ts

```diff
@@ -0,0 +1,67 @@
+export interface FlashMessage {
+  id: number
+  text: string
+  style?: React.HTMLAttributes<HTMLDivElement>['style']
+  classes?: string
+  persist?: boolean
+  viewed?: boolean
+}
+
+type FlashAction =
+  | {
+      type: 'ADD_MESSAGE'
+      message: Omit<FlashMessage, 'id'>
+    }
+  | {
+      type: 'DISMISS_MESSAGE'
+      messageId: number
+    }
+  | {
+      type: 'CYCLE_MESSAGE'
+      messageId: number
+    }
+
+// the reducer
+export default (messages: FlashMessage[] = [], action: FlashAction) => {
+  switch (action.type) {
+    case 'ADD_MESSAGE': {
+      const {
+        text,
+        style = {},
+        classes = '',
+        persist = false,
+        viewed = false,
+      } = action.message
+
+      const newMessage = {
+        id: messages.length,
+        text,
+        style,
+        classes,
+        persist,
+        viewed,
+      }
```

Comment:
```suggestion
      const newMessage = {
        ...action.message,
        id: messages.length,
        style: action.message.style || {},
      }
```

I'm not sure the default values are actually needed. TS will force you to check for `undefined` even with the default values, and then you might as well just handle it where you're using it.

Checks like `if (message.persist)` works with both `undefined` and `false`. So not much benefit of the default value.

Reply:
I like this line of reasoning